share/config.c

```diff
@@ -183,10 +183,10 @@ static struct
     { &CONFIG_JOYSTICK_BUTTON_L2,     "joystick_button_l2",     6 },
     { &CONFIG_JOYSTICK_BUTTON_R2,     "joystick_button_r2",     7 },
     { &CONFIG_JOYSTICK_BUTTON_START,  "joystick_button_start",  8 },
-    { &CONFIG_JOYSTICK_DPAD_L,        "joystick_dpad_l",       -1 },
-    { &CONFIG_JOYSTICK_DPAD_R,        "joystick_dpad_r",       -1 },
-    { &CONFIG_JOYSTICK_DPAD_U,        "joystick_dpad_u",       -1 },
-    { &CONFIG_JOYSTICK_DPAD_D,        "joystick_dpad_d",       -1 },
+    { &CONFIG_JOYSTICK_DPAD_L,        "joystick_dpad_l",        9 },
+    { &CONFIG_JOYSTICK_DPAD_R,        "joystick_dpad_r",        10 },
+    { &CONFIG_JOYSTICK_DPAD_U,        "joystick_dpad_u",        11 },
+    { &CONFIG_JOYSTICK_DPAD_D,        "joystick_dpad_d",        12 },
```

Comment:
D-pad buttons are really not on the same list because they come from a completely different API -
 not from SDL. I say leave it alone unless you can confirm that whatever indices you put there actually somewhat make sense for the device and the API.